lstchain/tools/lstchain_create_dl2_file.py

```diff
@@ -0,0 +1,161 @@
+"""
+Create h5 file with reconstruction of energy, disp and gamma/hadron separation of events.
+"""
+
+import joblib
+import numpy as np
+import pandas as pd
+from ctapipe.core import Provenance, Tool, traits
+from lstchain.io import (
+    get_dataset_keys,
+    read_configuration_file,
+    standard_config,
+    write_dl2_dataframe,
+)
+from lstchain.io.io import (
+    dl1_images_lstcam_key,
+    dl1_params_lstcam_key,
+    dl1_params_src_dep_lstcam_key,
+)
+from lstchain.reco import dl1_to_dl2
+from lstchain.reco.utils import filter_events, impute_pointing, add_delta_t_key
+from tables import open_file
+
+
+class ReconstructionHDF5Writer(Tool):
+
+    name = "ReconstructionHDF5Writer"
+    description = "Generate a HDF5 file with reconstructed energy, disp and hadroness of events"
+
+    input = traits.Path(
+        help="Path to a DL1 HDF5 file", directory_ok=False, exists=True
+    ).tag(config=True)
+    path_models = traits.Path(
+        help="Path where to find the trained RF models", file_ok=False
+    ).tag(config=True)
+    output_dir = traits.Path(
+        help="Path where to store the reconstructed DL2", file_ok=False
+    ).tag(config=True)
+    config_file = traits.Path(
+        help="Path to a configuration file. If none is given, a standard configuration is applied",
+        directory_ok=False,
+    ).tag(config=True)
+
+    aliases = {
+        ("i", "input"): "ReconstructionHDF5Writer.input",
+        ("o", "output"): "ReconstructionHDF5Writer.output_dir",
+        "models": "ReconstructionHDF5Writer.path_models",
+        "config": "ReconstructionHDF5Writer.config_file",
+    }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        """
+        Tool that generates a HDF5 file with reconstructed energy, disp and hadroness of events.
+
+        For getting help run:
+        lstchain_create_dl2_file --help
+        """
+        self.configuration = None
+        self.data = None
+        self.reg_energy = None
+        self.reg_disp_vector = None
+        self.cls_gh = None
+        self.dl2 = None
+
+    def setup(self):
```

Comment:
I think `setup` should only initialize components that need the configuration. Any actual work, like loading input data and so on should happen in `run`.

Reply:
No, the code checking the traitlets and so on belongs here, but I would move the loading of the data and the models.

Any actual work should be done in `start` rather than `setup`